azurerm/internal/services/kusto/kusto_iothub_data_connection_resource_test.go

```diff
@@ -31,7 +31,80 @@ func TestAccKustoIotHubDataConnection_basic(t *testing.T) {
 	})
 }
 
-func (KustoIotHubDataConnectionResource) basic(data acceptance.TestData) string {
+func TestAccKustoIotHubDataConnection_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_kusto_iothub_data_connection", "test")
+	r := KustoIotHubDataConnectionResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (KustoIotHubDataConnectionResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := parse.DataConnectionID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Kusto.DataConnectionsClient.Get(ctx, id.ResourceGroup, id.ClusterName, id.DatabaseName, id.Name)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %v", id.String(), err)
+	}
+
+	value, ok := resp.Value.AsIotHubDataConnection()
+	if !ok {
+		return nil, fmt.Errorf("%s is not an IotHub Data Connection", id.String())
+	}
+
+	return utils.Bool(value.IotHubConnectionProperties != nil), nil
+}
+
+func (r KustoIotHubDataConnectionResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_kusto_iothub_data_connection" "test" {
+  name                = "acctestkedc-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  cluster_name        = azurerm_kusto_cluster.test.name
+  database_name       = azurerm_kusto_database.test.name
+
+  iothub_id                 = azurerm_iothub.test.id
+  consumer_group            = azurerm_iothub_consumer_group.test.name
+  shared_access_policy_name = azurerm_iothub_shared_access_policy.test.name
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r KustoIotHubDataConnectionResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_kusto_iothub_data_connection" "test" {
+  name                = "acctestkedc-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  cluster_name        = azurerm_kusto_cluster.test.name
+  database_name       = azurerm_kusto_database.test.name
+
+  iothub_id                 = azurerm_iothub.test.id
+  consumer_group            = azurerm_iothub_consumer_group.test.name
+  shared_access_policy_name = azurerm_iothub_shared_access_policy.test.name
+  event_system_properties   = ["message-id", "sequence-number", "to"]
+  mapping_rule_name = "Json_Mapping"
+  data_format       = "MULTIJSON"
```

Comment:
I believe the tflint error is coming from here

```suggestion
  mapping_rule_name         = "Json_Mapping"
  data_format               = "MULTIJSON"
```

Reply:
Thanks, I had to just install terrafmt :)

Reply:
I've gotten all the terraform linting issues, but now I'm afraid there will be markdown linting that I don't know about 😆